File: week10/in-class-exercises/python/q1a.py
# Nicholas Ong - Q1(a) Numbers in Files
def sum_up(input_file_name):
    infile = open(f"../src/{input_file_name}", "r")
    result = sum([float(i.strip()) for i in infile.readlines()])
    print(result)
    infile.close()

def main():
    test_cases = ["q1-1.txt", "q1-2.txt"]
    for test in test_cases:
        sum_up(test)

if __name__ == "__main__":
    main()

File: week10/in-class-exercises/cpp/q1a.cpp
#include <iostream>
#include <fstream>
#include <string>

void sum_up(std::string filestring)
{
  std::fstream infile;
  infile.open(filestring,std::ios::in);

  float sum = 0;
  std::string line;

  while (std::getline(infile,line))
  {
    sum += atof(line.c_str());
  }

  std::cout << sum << std::endl;
}

int main()
{
  sum_up("../src/q1-1.txt");
  sum_up("../src/q1-2.txt");
}
